spacy/cli/project/parallel.py

```diff
@@ -0,0 +1,442 @@
+from typing import Any, List, Optional, Dict, Union, cast
+import os
+import sys
+from pathlib import Path
+from time import time
+from multiprocessing import Manager, Queue, Process, get_context
+from multiprocessing.context import SpawnProcess
+from shutil import rmtree
+from signal import SIGTERM
+from subprocess import STDOUT, Popen, TimeoutExpired
+from wasabi import msg
+from wasabi.util import color, supports_ansi
+
+from .._util import check_rerun, check_deps, update_lockfile, load_project_config
+from ...util import SimpleFrozenDict, working_dir, split_command, join_command
+from ...util import check_bool_env_var, ENV_VARS
+from ...errors import Errors
+
+"""
+Permits the execution of a parallel command group.
+
+The main process starts a worker process for each spaCy projects command in the parallel group.
+Each spaCy projects command consists of n OS level commands. The worker process starts
+subprocesses for these OS level commands serially, and sends status information about their
+execution back to the main process.
+
+The main process maintains a state machine for each spaCy projects command/worker process. The meaning
+of the states is documented alongside the _PARALLEL_COMMAND_STATES code. Note that the distinction
+between the states 'failed' and 'terminated' is not meaningful on Windows, so that both are displayed
+as 'failed/terminated' on Windows systems.
+"""
+
+# Use spawn to create worker processes on all OSs for consistency
+mp_context = get_context("spawn")
+
+# How often the worker processes managing the commands in a parallel group
+# send keepalive messages to the main process (seconds)
+PARALLEL_GROUP_STATUS_INTERVAL = 1
+
+# The dirname where the temporary logfiles for a parallel group are written
+# before being copied to stdout when the group has completed
+PARALLEL_LOGGING_DIR_NAME = "parrTmp"
+
+# The maximum permissible width of divider text describing a parallel command group
+MAX_WIDTH_DIVIDER = 60
+
+# Whether or not to display realtime status information
+DISPLAY_STATUS_TABLE = sys.stdout.isatty() and supports_ansi()
```

Comment:
If there status table cannot be displayed in real time, it isn't displayed at all because it wouldn't serve any useful purpose. If ANSI is not available, nothing is displayed unless the `parallel[..., ...]` description was too wide to be displayed in the divider, in which case it is outputted again in full.

Reply:
Or do you feel it would make sense to display the status table a single time once the group has completed?

Reply:
No, I think not displaying it at all if it can't be shown in real time is reasonable.